client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminCart.tsx

```diff
@@ -80,14 +145,67 @@ const AdminCart = ({
           Add more items
         </button>
       </section>
-      <aside className='flex flex-col items-center gap-2 bg-zinc-800 rounded-xl px-5 py-3'>
-        <div className='flex flex-row items-center gap-2 justify-between w-full'>
-          <span className='text-zinc-100 text-sm '>Total</span>
+      <section className='flex flex-col items-center gap-2 bg-zinc-800 rounded-xl px-5 py-3'>
+        <div className={
+          `bg-zinc-700 flex items-center justify-center gap-1 p-1 rounded-lg shadow-md w-full
+          ${appliedDiscount && 'bg-zinc-800 border-2 border-zinc-900'}`
+        }>
+          <input
+            type='text'
+            placeholder='Discount code...'
+            aria-label='Discount code'
+            className='p-1 px-2 w-full rounded-md text-zinc-100 bg-zinc-700 disabled:bg-zinc-800'
+            value={
+              discountText
+                ? discountText
+                : appliedDiscount
+                ? appliedDiscount.code
+                : ''
+            }
+            onChange={(e) => {
+              setDiscountText(e.target.value);
+              setDiscountClicked(false);
+            }}
+            disabled={appliedDiscount}
+          />
+          {!appliedDiscount ? (
+            <button
+              className='p-2 text-zinc-400 hover:bg-zinc-900 hover:text-zinc-300 justify-end rounded-full'
+              onClick={handleApplyDiscount}
+              aria-label='Apply discount code'
+            >
+              <SearchIcon className='h-5 w-5' strokeWidth={3} />
+            </button>
+          ) : (
+            <button
+              className='p-2 text-zinc-400 hover:bg-zinc-900 hover:text-zinc-300 justify-end rounded-full'
+              onClick={handleRemoveDiscount}
+              aria-label='Remove discount code'
+            >
+              <XIcon className='h-5 w-5' strokeWidth={3} />
+            </button>
+          )}
+        </div>
+        {discountClicked &&
+          (!appliedDiscount ? (
+            <p className='bg-amber-400 text-sm font-medium text-zinc-900 rounded-md px-3'>
+              Invalid Discount Code
+            </p>
+          ) : (
+            <aside className='flex items-center gap-2 justify-between w-full'>
+              <h2 className='text-zinc-100 text-sm'>Discount</h2>
+              <strong className='text-amber-300 text-lg font-bold'>
+                {toDollar(subtotal - getTotal())}
+              </strong>
+            </aside>
+          ))}
+        <aside className='flex items-center gap-2 justify-between w-full'>
+          <h2 className='text-zinc-100 text-sm'>Total</h2>
           <strong className='text-white text-lg font-bold'>
-            {toDollar(total)}
+            {toDollar(getTotal())}
```

Comment:
Not a big deal because it doesn't effect the backend, but if the discount amount is greater than the cart amount the subtotal goes negative. I think line 89 just needs to take the max of 0 or the current calculation.

Reply:
Thanks for the catch! Fixed